pkg/cache/context.go

```diff
@@ -295,41 +295,41 @@ func (ctx *Context) UpdatePod(_, newObj interface{}) {
 }
 
 func (ctx *Context) updateYuniKornPod(appID string, pod *v1.Pod) {
-	var app *Application
 	taskID := string(pod.UID)
-	if app = ctx.getApplication(appID); app != nil {
+	app := ctx.getApplication(appID)
+	if app != nil {
 		if task := app.GetTask(taskID); task != nil {
 			task.setTaskPod(pod)
 		}
 	}
 
 	// treat terminated pods like a remove
 	if utils.IsPodTerminated(pod) {
-		ctx.notifyTaskComplete(appID, taskID)
+		if app != nil {
```

Comment:
Perhaps we could move the condition `if app != nil ` back to `notifyTaskComplete` and simplify `deleteYuniKornPod`
from
```go
                 if app := ctx.GetApplication(taskMeta.ApplicationID); app != nil {
			ctx.notifyTaskComplete(taskMeta.ApplicationID, taskMeta.TaskID)
		}
```
to 
```go
		ctx.notifyTaskComplete(ctx.GetApplication(taskMeta.ApplicationID), taskMeta.TaskID)
```

WDYT ?

Reply:
I'm with this change. That way, it change the behaviour of `notifyTaskComplete()` from assuming app is not nil to if nil occur it will return.

Reply:
I guess `if app != nil {` can be removed now, right?

Reply:
You're right! Done.